test/factories/coral_cover.jl

```diff
@@ -0,0 +1,60 @@
+using ADRIA: YAXArray, DataCube, ZeroDataCube
+
+"""Raw model coral cover results, proportional to k-area."""
+function coral_cover(;
+    n_timesteps::Int64=10,
+    n_group_sizes::Int64=35,
+    n_locations::Int64=5,
+    n_scenarios::Int64=32
+)::YAXArray{Float64,4}
+    coral_cover = Factories.full_coral_cover(;
+        n_timesteps, n_group_sizes, n_locations, n_scenarios
+    )
```

Comment:
`Factories` is not a great name, but also, is this exported directly from ADRIA? I don't see how it's imported for use here.

Reply:
I've used Factories inspired by Factories in Ruby [1]. But I'm happy to change to something else. Any suggestions?

[1] https://stackoverflow.com/questions/37364454/what-is-a-ruby-factory-method

Reply:
About how it is imported, I'm including it in `runmodel.jl`.

Reply:
> I've used Factories inspired by Factories in Ruby [1]. But I'm happy to change to something else. Any suggestions?
> 
> [1] https://stackoverflow.com/questions/37364454/what-is-a-ruby-factory-method

Yes, I'm familiar with the factory pattern (it's not specific to Ruby by the way).

The current name is too general. In the example linked above they name it `PersonFactory` which at least indicates what the factory is for, but even that is for an example. Equally, (I hope) you'd agree `PersonConstructor` is not a great name.

What does the factory do?
From what I can see it creates mock coral cover data, so this specific use could be `mock_full_coral_cover`.

Some more questions:

1. Where is `runmodel.jl`? I don't see it in the list of files modified by this PR
2. I would call `full_coral_cover` -> `maximum_coral_cover`. Or at least explain more in the docstring ("represent condition where all locations have maximum possible coral cover").

Reply:
You said `runmodel.jl`

> About how it is imported, I'm including it in `runmodel.jl`.

Reply:
I agree that `Factories` is too vague. If this is for generating mock data, then why not a name like `MockData`?
`MockData.full_coral_cover`?
I think it would be cleaner to have more modules with meaningful names rather than putting lots of misc things into Factories.

As far as static constructor functions, I've seen "build" or "create" in names. e.g. `Person.create`, so if you prefer:
`MockData.create_full_coral_cover`

or `mock_full_coral_cover` as Takuya suggested.

Reply:
I don't mind switching to `MockData`. I don't like using "create" because all functions in this module will be to create mock data so all of them will have this prefix and that will make the function names too big. 

Do we agree with the module being called `MockData` then?